.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,150 @@
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      JAVA_VERSION: '21'
+      CONFIG_DIR: '.ci-config'
+      DIFFTOOL_JAR: "diff-java-tool.jar"
+      PATCH_DIFF_TOOL_VERSION: "0.1-SNAPSHOT"
+      CI_TEMP_DIR: ".ci-temp"
+
+    steps:
+      - name: Checkout Main Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup .ci-temp Directory
+        run: |
+          mkdir -p "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}"
+
+      - name: Checkout Checkstyle Repository
+        uses: actions/checkout@v4
+        with:
+          repository: checkstyle/checkstyle
+          path: "${{ env.CI_TEMP_DIR }}/checkstyle"
+
+      - name: Set up JDK ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.JAVA_VERSION }}
+          cache: 'gradle'
+
+      - name: Cache Maven Dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-
+
+      - name: Build with Gradle
+        run: |
+          cd "${{ github.workspace }}/diff-java-tool"
+          ./gradlew clean build
+
+      - name: Copy DiffTool JAR to .ci-temp
+        run: |
+          cp "${{ github.workspace }}/diff-java-tool/build/libs/diff-java-tool-*-all.jar" "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}/${{ env.DIFFTOOL_JAR }}"
+
+      - name: Download patch-diff-report-tool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "checkstyle/contribution"
+          tag: "patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}"
+          fileName: >-
+            patch-diff-report-tool-${{ env.PATCH_DIFF_TOOL_VERSION }}-jar-with-dependencies.jar
+          out-file-path: "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}"
+
+      - name: Create Temporary Patch Branch
+        run: |
+          cd "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}/checkstyle"
+          git config user.email "github-actions@github.com"
+          git config user.name "GitHub Actions"
+          git checkout -b temp-patch-branch
+          # Optionally make a minor change to ensure reports are different
+          echo "// Temporary change" >> dummy.java
+          git add dummy.java
+          git commit -m "Temporary change for testing"
+
+      - name: Initialize git repository in 'diff-java-tool'
+        run: |
+          mkdir -p "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}/diff-java-tool"
+          cd "${{ github.workspace }}/${{ env.CI_TEMP_DIR }}/diff-java-tool"
+          git init --initial-branch=main
+          git config user.email "github-actions@github.com"
+          git config user.name "GitHub Actions"
+          git add .
+          git commit -m "Initialize diff-java-tool repository"
+
+      - name: Execute DiffTool.jar with config
```

Comment:
.... diff-java-tool jar ....

Reply:
same for others

Reply:
done

Reply:
not done, please recheck push

Reply:
`- name: Execute diff-java-tool.jar with config`

Reply:
done